pi/mlir/utils.py

```diff
@@ -173,13 +173,27 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def random_normal_wrapper(factory):
+    @functools.wraps(factory)
+    def wrapper(*args, **kwargs):
+        # Create a random number generator
+        rng = np.random.default_rng()
+        if len(args) == 1 and isinstance(args[0], (list, tuple)):
+            return factory(rng, args[0], **kwargs)
+        else:
+            return factory(rng, args, **kwargs)
+
+    return wrapper
```

Comment:
what you're trying to do here is emulate `star_args_wrapper` because you can't just straight up reuse it because you need `rng`.

what you should do is write the case for non-star args; something like 

```python
def standard_normal(...)
    rng = np.random.default_rng()
    np.random.Generator.standard_normal(rng, ...)
```

and then `standard_normal` in `star_args_wrapper`, i.e. 
   
```python
randn = functools.partial(_np_wrapper, factory=star_args_wrapper(standard_normal))
```